components/sup/src/manager.rs

```diff
@@ -1894,14 +1894,16 @@ fn get_fd_count() -> std::io::Result<usize> {
     }
 }
 
-// Assume we have /proc/self/fd unless we know we don't
-#[cfg(not(any(target_os = "freebsd", target_os = "macos", target_os = "ios")))]
-const FD_DIR: &str = "/proc/self/fd";
-#[cfg(any(target_os = "freebsd", target_os = "macos", target_os = "ios"))]
-const FD_DIR: &str = "/dev/fd";
-
 #[cfg(unix)]
-fn get_fd_count() -> std::io::Result<usize> { Ok(fs::read_dir(FD_DIR)?.count()) }
+fn get_fd_count() -> std::io::Result<usize> {
+    // Assume we have /proc/self/fd unless we know we don't
+    #[cfg(not(any(target_os = "freebsd", target_os = "macos", target_os = "ios")))]
```

Comment:
I _think_ `openbsd` and `android` are likely to have /proc/self/fd as well.  On the other hand, I kinda wonder if we could just _always_ use `/dev/fd` as linux will usually have it as well.

FWIW, I went with these loosey-goosey constrains simply because the old library we were using would have err'd out anyway on any platform that isn't freebsd, macos, ios, android, or linux.  So, definitely some more though/research could be done here.

Reply:
The intent of this change was just to fix an unused variable compiler warning on windows. It just moves the `FD_DIR` declaration from outside the `get_fd_count` function to inside (`get_fd_count` is only defined for unix OSs). Here is the [library](https://github.com/alecmocatta/palaver/blob/07194c645842644f2464fbdb157c516216acf6ea/src/file.rs#L552) where the logic is taken from. I am tempted to either: leave the change as is or if we feel the logic is wrong I can open up a separate PR to fix it.

Reply:
Definitely +1 on doing any further code changes to this in a different PR.

Reply:
Ok with not fixing that here.